include/hal/library/responder/setcertlib.h

```diff
@@ -16,16 +16,23 @@
 /**
  * return if current code is running in a trusted environment.
  *
+ * @param[in]  spdm_context  A pointer to the SPDM context.
+ *
  * @retval  true   It is in a trusted environment.
  * @retval  false  It is not in a trusted environment.
  **/
-extern bool libspdm_is_in_trusted_environment();
+extern bool libspdm_is_in_trusted_environment(
+#if LIBSPDM_HAL_PASS_SPDM_CONTEXT
+    void *spdm_context
```

Comment:
This one probably doesn't need it since it applies to all contexts. Ie, the device is in a secure manufacturing environment or not.

Reply:
I understand what you're saying, in my case I'd use it so I can access callbacks that would tell me the actual state. Otherwise I'd have to make this a global. So it's less about it being context specific, but more about allowing me to access outside helpers to get the result without hardcoding global variables. But I can understand you being hesitant to this.

Reply:
I don't mind to simply add `spdm_context` to all APIs, just in case someone wants it.

Reply:
Yeah that's fine.